test/new-e2e/tests/apm/tests.go

```diff
@@ -24,6 +24,8 @@ import (
 	"github.com/stretchr/testify/suite"
 )
 
+// Add new e2e test for APM IOT mode as well
```

Comment:
We should implement this as part of the PR. You can look into the `testBasicTraces` function and perhaps extend it to ensure payloads contain `APMMode==full`, and optionally add another test where the env var is injected.